.github/workflows/CI.yml

```
@@ -793,6 +794,11 @@ jobs:
           mkdir -p artifacts
           cd build
 
+          # save the original binaries with debug info
+          for f in $(find . -name '*.exe' -not -path './CMakeFiles/*'); do
+            zip ../artifacts/sunshine-debuginfo-win32.zip "$f"
+          done
+
```

```suggestion
          # save the original binaries with debug info
          7z \
            "-xr!CMakeFiles" \
            a "../artifacts/sunshine-debuginfo-win32.zip" "*.exe"

```

This wouldn't require installing another dependency. I believe 7z is included in all GitHub runners by default.

---

Also, FYI... there's just one artifact uploaded... all 3 artifacts will reside in the `sunshine-windows.zip` artifact.

- sunshine-windows-installer.exe
- sunshine-windows-portable.zip
- sunshine-debuginfo-win32.zip

---

> This wouldn't require installing another dependency. I believe 7z is included in all GitHub runners by default.

It might be included in a normal shell, but it doesn't seem to be on MSYS2's `PATH`. https://github.com/cgutman/LB_Sunshine/actions/runs/4904317553/jobs/8757283657

> Also, FYI... there's just one artifact uploaded... all 3 artifacts will reside in the sunshine-windows.zip artifact.

Yep, that's true for branches, but nightly/release should publish them individually right? Or do I need to make another change to do that?

---

> should publish them individually right

For releases, each file (all 3) in the artifacts directory will be a separate release asset.

In that case, I suggest to make a separate step, which will resolve both issues (assuming you don't want to upload the debug info as a release asset).

```yml
- name: Package Windows Debug
  if: ${{ needs.setup_release.outputs.create_release != 'true' }}
  working-directory: build
  run: |
    # save the original binaries with debug info
    7z \
      "-xr!CMakeFiles" \
      "-xr!cpack_artifacts" \
      a "../artifacts/sunshine-debuginfo-win32.zip" "*.exe"
```

If you do want it uploaded with releases, just remove the `if` line.

---

We do want it for releases because we'll need it to debug crashes in those builds (which will probably be what most users run).